lib/rspec/core/option_parser.rb

```diff
@@ -22,8 +22,12 @@ def parse(source=nil)
       begin
         parser(options).parse!(args)
       rescue OptionParser::InvalidOption => e
-        failure = e.message
-        failure << " (defined in #{source})" if source
+        failure =
+          if source
+            "#{e.message} (defined in #{source})"
+          else
+            e.message
+          end
```

Comment:
Maybe:
```ruby
source_location = " (defined in #{source})" if source
abort "#{e.message}#{source_location}\n\n...
```

Reply:
Refactored a bit

Reply:
Can you take a look at rspec/rspec-support#392